docs/openstack/edpm_adoption.md

```diff
@@ -95,6 +95,18 @@ EOF
   $(cat ~/install_yamls/out/edpm/ansibleee-ssh-key-id_rsa | base64 | sed 's/^/        /')
   EOF
   ```
+* Create the Nova Cell secret (Workaround while nova isn't adopted yet):
+
+  ```bash
+  oc apply -f - <<EOF
+  apiVersion: v1
+  kind: Secret
+  metadata:
+      name: nova-metadata-neutron-config
+  data:
+      05-nova-metadata.conf: $("[DEFAULT]\nnova_metadata_host = http://nova-metadata-internal.openstack.svc:8775\nmetadata_proxy_shared_secret = 1234567842\n" | base64)
```

Comment:
I don't think this is correct, the format is

```
[DEFAULT]
nova_metadata_host = 1.2.3.4
nova_metadata_port = 8775
nova_metadata_protocol = http
metadata_proxy_shared_secret = 12345678
```

Reply:
I got it from a running env:
```
[dpadev@osp-dev-05 devsetup]$ oc get secret nova-metadata-neutron-config -o json | jq '.data | map_values(@base64d)' 
{
  "05-nova-metadata.conf": "[DEFAULT]\nnova_metadata_host = http://nova-metadata-internal.openstack.svc:8775\nmetadata_proxy_shared_secret = 1234567842\n",
  "httpd.conf": "ServerTokens Prod\nServerSignature Off\nTraceEnable Off\nPidFile run/httpd.pid\nServerRoot \"/etc/httpd\"\nServerName \"localhost.localdomain\"\n\nUser apache\nGroup apache\n\nListen 8775\n\nTypesConfig /etc/mime.types\n\nInclude conf.modules.d/*.conf\n\nLogFormat \"%h %l %u %t \\\"%r\\\" %>s %b \\\"%{Referer}i\\\" \\\"%{User-Agent}i\\\"\" combined\nLogFormat \"%{X-Forwarded-For}i %l %u %t \\\"%r\\\" %>s %b \\\"%{Referer}i\\\" \\\"%{User-Agent}i\\\"\" proxy\n\nSetEnvIf X-Forwarded-For \"^.*\\..*\\..*\\..*\" forwarded\nCustomLog /dev/stdout combined env=!forwarded\nCustomLog /dev/stdout proxy env=forwarded\n## set default apache log level to info from warning\nLogLevel info\n\n<VirtualHost *:8775>\n  <IfVersion >= 2.4>\n    ErrorLogFormat \"%M\"\n  </IfVersion>\n  SetEnvIf X-Forwarded-For \"^.*\\..*\\..*\\..*\" forwarded\n  ErrorLog /dev/stdout\n  CustomLog /dev/stdout combined env=!forwarded\n  CustomLog /dev/stdout proxy env=forwarded\n  ## set nova vhost log level to debug\n  LogLevel debug\n\n  ## WSGI configuration\n  WSGIProcessGroup nova-metadata\n  WSGIApplicationGroup %{GLOBAL}\n  WSGIPassAuthorization On\n  ## In general we want nova-metadata to scale via k8s replicas but we need\n  ## two processes per replica to always has a room for a healthecheck query\n  WSGIDaemonProcess nova-metadata processes=2 threads=1 user=nova group=nova display-name=nova-metadata-api\n  WSGIScriptAlias / /usr/bin/nova-metadata-wsgi\n</VirtualHost>\n\nAlias /nova-metadata /usr/bin/nova-metadata-wsgi\n<Location /nova-metadata>\n  SetHandler wsgi-script\n  Options +ExecCGI\n  WSGIProcessGroup nova-metadata\n  WSGIApplicationGroup %{GLOBAL}\n  WSGIPassAuthorization On\n</Location>\n",
  "nova-metadata-config.json": "{\n  \"command\": \"/usr/sbin/httpd -DFOREGROUND\",\n  \"config_files\": [\n    {\n      \"source\": \"/var/lib/openstack/config/nova-blank.conf\",\n      \"dest\": \"/etc/nova/nova.conf\",\n      \"owner\": \"nova\",\n      \"perm\": \"0600\"\n  },\n  {\n      \"source\": \"/var/lib/openstack/config/01-nova.conf\",\n      \"dest\": \"/etc/nova/nova.conf.d/01-nova.conf\",\n      \"owner\": \"nova\",\n      \"perm\": \"0600\"\n  },\n  {\n      \"source\": \"/var/lib/openstack/config/02-nova-override.conf\",\n      \"dest\": \"/etc/nova/nova.conf.d/02-nova-override.conf\",\n      \"owner\": \"nova\",\n      \"perm\": \"0600\",\n      \"optional\": true\n  },\n    {\n      \"source\": \"/var/lib/openstack/config/httpd.conf\",\n      \"dest\": \"/etc/httpd/conf/httpd.conf\",\n      \"owner\": \"apache\",\n      \"perm\": \"0644\"\n    }\n  ],\n  \"permissions\": [\n    {\n      \"path\": \"/var/log/nova\",\n      \"owner\": \"nova:apache\",\n      \"recurse\": true\n    }\n  ]\n}\n"
}
```
and the CI got green

Reply:
It got "green" because ansibleee role was executed and the container for the agent was created. But if you look at the logs for the agent, you will see that it crashes over and over because it can't parse the config option value for nova_metadata_host (that should be a host, without protocol / port). I am not sure where to find the logs for EDP service containers in adoption CI artifacts though. Do you have an idea?

Reply:
it is under "controller": https://logserver.rdoproject.org/57/157/fbb5a720a18ee16a2c442aa0686de2e4302e20bc/github-check/data-plane-adoption-github-rdo-centos-9-crc-single-node/d89f7af/controller/pod/

Reply:
As to the fact that you got it from a running env, this is because there's a bug (fixed now) in nova-operator, see https://github.com/openstack-k8s-operators/nova-operator/pull/534

Reply:
gotcha!
I'll update the PR

Reply:
> it is under "controller"
I am not sure. These seem like pod logs. I am talking about ovn_metadata_agent container running on EDP nodes (not on CP nodes).

Reply:
yes, https://review.rdoproject.org/r/c/rdo-jobs/+/50082 was wrong and didn't catch those logs, I've updated it now
we will have the logs in the next run

Reply:
@fao89 this is helpful, but I think it still won't capture the logs for actual containers that run services like ovn-controller or metadata-agent on EDP nodes. Your patch will collect logs from pods that run operators, not from containers running on EDP.

Reply:
do you want something like this? https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/artifacts/tasks/edpm.yml#L58

Reply:
@fao89 yes, this would do.

Reply:
@cescgina do you know how can we collect these logs from the standalone node?

Reply:
@fao89 we could add an extra task here https://review.rdoproject.org/r/plugins/gitiles/rdo-jobs/+/refs/heads/master/playbooks/data_plane_adoption/standalone_collect_logs.yaml to run the needed comands and pull the logs out of the standalone vm/node